src/client.rs

```diff
@@ -109,10 +104,9 @@ impl Client {
     /// # });
     /// ```
     pub async fn get_index(&self, uid: impl AsRef<str>) -> Result<Index, Error> {
-        match self.get_raw_index(uid).await {
-            Ok(raw_idx) => Ok(raw_idx.into_index(self)),
-            Err(error) => Err(error),
-        }
+        let mut idx = self.index(uid.as_ref());
```

Comment:
@irevoire this needs to be mutable?

Reply:
Yes, the `Index::fetch_info` method takes a mutable reference to self, as it needs to update self.